config.go

```diff
@@ -159,57 +173,90 @@ func (ca *Cagent) userAgent() string {
 
 func (ca *Cagent) DumpConfigToml() string {
 	buff := &bytes.Buffer{}
-	enc := toml.NewEncoder(buff)
-	err := enc.Encode(ca)
 
+	err := toml.NewEncoder(buff).Encode(ca)
 	if err != nil {
 		log.Errorf("DumpConfigToml error: %s", err.Error())
+		return ""
 	}
 
 	return buff.String()
 }
 
-func (ca *Cagent) ReadConfigFromFile(configFilePath string, createIfNotExists bool) error {
-	dir := filepath.Dir(configFilePath)
-	err := os.MkdirAll(dir, 0755)
+func (ca *Cagent) ReadConfigFromFile(configFilePath string) error {
+	_, err := os.Stat(configFilePath)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to create the config dir: '%s'", dir)
+		return err
 	}
 
-	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
-		if !createIfNotExists {
-			return fmt.Errorf("Config file not exists: %s", configFilePath)
-		}
-		f, err := os.OpenFile(configFilePath, os.O_WRONLY|os.O_CREATE, 0644)
+	_, err = toml.DecodeFile(configFilePath, &ca)
 
-		if err != nil {
-			return fmt.Errorf("Failed to create the default config file: '%s'", configFilePath)
-		}
-		defer f.Close()
-		enc := toml.NewEncoder(f)
-		enc.Encode(ca)
-	} else {
-		_, err = os.Stat(configFilePath)
-		if err != nil {
-			return err
+	ca.lookupEnv()
```

Comment:
The function is called `ReadConfigFromFile`. So it should not read something from environment. The name is misleading then

Reply:
common approach is to to rewrite values with environment if they are provided
Might be function should be renamed to ```ReadConfig```

Reply:
Simply renaming does not solve the problem.
A function should (ideally) do _one_ thing. As said in the other comment, why not call `lookupEnv ()` before this function is called?